python/lsst/ip/diffim/diffimTools.py

```diff
@@ -307,7 +307,10 @@ def sourceToFootprintList(candidateInList, templateExposure, scienceExposure, ke
 
     for kernelCandidate in candidateInList:
         if not type(kernelCandidate) == afwTable.SourceRecord:
-            raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
+            try:
+                kernelCandidate = kernelCandidate.getSource()
+            except:
+                raise RuntimeError, ("Candiate not of type afwTable.SourceRecord")
```

Comment:
Use the modern version of `raise`:

    raise RuntimeError("Candidate something something")